### PR TITLE
Chapter 9: Advanced Login

### PR DESCRIPTION
**Pull request by baaohuii**

### CODE REVIEW

**Comment by baaohuii** [comment]
ready

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -2,17 +2,38 @@ module SessionsHelper
...
+        @current_user = user
+      end
+    end
   end
 
   def logged_in?
     current_user.present?
   end
 
+  def forget(user)
```
Comment (ngocvt-0484) [inline app/helpers/sessions_helper.rb]: remove ( )

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -2,17 +2,38 @@ module SessionsHelper
...
+    user.forget
+    cookies.delete(:user_id)
```
Comment (ngocvt-0484) [inline app/helpers/sessions_helper.rb]: remove ( )

**Comment by baaohuii** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 3]
```diff
@@ -129,3 +129,17 @@ input.btn.btn-primary {
     color: $state-danger-text;
   }
 }
+
+.checkbox {
+  margin-top: -10px;
+  margin-bottom: 10px;
+  span {
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]:
```suggestion
  margin-bottom: 10px;
  
  span {
```

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -2,9 +2,10 @@ class SessionsController < ApplicationController
   def new; end
 
   def create
-    user = User.find_by email: params[:session][:email].downcase
-    if user&.authenticate params[:session][:password]
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user && user.authenticate(params[:session][:password])
```
Comment (saobv-2223) [inline app/controllers/sessions_controller.rb]: remove (), please

File: app/controllers/sessions_controller.rb [hunk 5]
```diff
@@ -2,9 +2,10 @@ class SessionsController < ApplicationController
...
       log_in user
+      params[:session][:remember_me] == '1' ? remember(user) : forget(user)
```
Comment (saobv-2223) [inline app/controllers/sessions_controller.rb]: move 1 to settings.yml file and call here

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -19,6 +19,10 @@ def create
     end
   end
 
+  def edit
+    @user = User.find(params[:id])
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]: remove ()

File: app/helpers/sessions_helper.rb [hunk 7]
```diff
@@ -2,16 +2,37 @@ module SessionsHelper
   def log_in user
     session[:user_id] = user.id
   end
+  
+  def remember(user)
```
Comment (saobv-2223) [inline app/helpers/sessions_helper.rb:6]: remove ()

File: app/helpers/sessions_helper.rb [hunk 8]
```diff
@@ -2,16 +2,37 @@ module SessionsHelper
...
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
 
   def current_user
-    @current_user ||= User.find_by id: session[:user_id]
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
```
Comment (saobv-2223) [inline app/helpers/sessions_helper.rb]: remove ()

File: app/helpers/sessions_helper.rb [hunk 9]
```diff
@@ -2,16 +2,37 @@ module SessionsHelper
...
   end
 
+  def forget user
+    user.forget
+    cookies.delete :user_id
+    cookies.delete :remember_token
+  end
+  
   def log_out
+    forget(current_user)
```
Comment (saobv-2223) [inline app/helpers/sessions_helper.rb]: remove ()

**Comment by baaohuii** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**